src/qml/filters/deband/ui.qml

```diff
@@ -0,0 +1,440 @@
+            filter.set(couplingParam, false)
+            filter.savePreset(allParams, qsTr('Full range to Limited range'))
+
+            // Default preset
+            // Same as Full to Limited preset
+            filter.savePreset(allParams)
+        }
+        filter.blockSignals = false
+
+        setControls()
+    }
+
+
+    width: 350
+    height: 450
+
+
+    GridLayout {
+        columns: 3
+        anchors.fill: parent
+        anchors.margins: 8
+
+        // Row split
+
+        Label {
+            text: qsTr('Preset')
+            Layout.alignment: Qt.AlignRight
+        }
+        Shotcut.Preset {
+            id: idPreset
+            Layout.columnSpan: 2
+            parameters: allParams
+            onPresetSelected: setControls()
+        }
+
+        // Row split
+
+        Label {
+            text: qsTr('Y/R Threshold')
+            Layout.alignment: Qt.AlignRight
+        }
+        Shotcut.SliderSpinner {
+            id: idThr1
+            minimumValue: 0
+            maximumValue: 100
+            decimals: 1
+            suffix: ' %'
+            onValueChanged: setThreshold(thr1Param, value)
+        }
+        Shotcut.UndoButton {
+            onClicked: idThr1.value = thrToPct(thr1Default)
+        }
+
+        // Row split
+
+        Label {
+            text: qsTr('Cb/G Threshold')
+            Layout.alignment: Qt.AlignRight
+        }
+        Shotcut.SliderSpinner {
+            id: idThr2
+            minimumValue: 0
+            maximumValue: 100
+            decimals: 1
+            suffix: ' %'
+            onValueChanged: setThreshold(thr2Param, value)
+        }
+        Shotcut.UndoButton {
+            onClicked: idThr2.value = thrToPct(thr2Default)
+        }
+
+        // Row split
+
+        Label {
+            text: qsTr('Cr/B Threshold')
+            Layout.alignment: Qt.AlignRight
+        }
+        Shotcut.SliderSpinner {
+            id: idThr3
+            minimumValue: 0
+            maximumValue: 100
+            decimals: 1
+            suffix: ' %'
+            onValueChanged: setThreshold(thr3Param, value)
+        }
+        Shotcut.UndoButton {
+            onClicked: idThr3.value = thrToPct(thr3Default)
+        }
+
+        // Row split
+
+        Label {
+            text: qsTr('Alpha Threshold')
+            Layout.alignment: Qt.AlignRight
+        }
+        Shotcut.SliderSpinner {
+            id: idThr4
+            minimumValue: 0
+            maximumValue: 100
+            decimals: 1
+            suffix: ' %'
+            onValueChanged: setThreshold(thr4Param, value)
+        }
+        Shotcut.UndoButton {
+            onClicked: idThr4.value = thrToPct(thr4Default)
+        }
+
+        // Row split
+
+        Label {}
+        CheckBox {
+            id: idLink
+            text: qsTr('Link thresholds')
+            onClicked: filter.set(linkParam, checked)
+        }
+        Label {}
+
+        // Filler
+
+        Label {
+            Layout.columnSpan: 3
```

Comment:
Are the fillers necessary? This could cause more people to require vertical scrolling depending on their screen resolution and panel layout. Maybe consider removing this filler and the one below.

Reply:
Agreed on vertical scrolling.  However, the reasoning was that the Randomize checkboxes were directly associated with the sliders above them.  If I put two more checkboxes below the lower Randomize checkbox with no filler, it could seem implied that those two checkboxes (Blur and All) were associated with the Slider as well.  But they're not.  They're independent of the sliders, and the filler was a way to imply disassociation.  I could see collapsing the upper filler.  But what are your thoughts on the middle filler between Randomize and Blur?

Reply:
My advice would be to word the labels in a way that the text creates the association you are looking for. For example, "Randomize between zero and value" could be changed to "Randomize pixel range between zero and value". And the next one could be "Randomize direction between zero degrees and value".

For complex filters like this one, people will need to learn the settings through study and trial-and-error. So I don't know how much the grouping really helps. Hopefully people will suggest improvements to the labels as they start to understand the filter.

Reply:
Done.